test/spec/controllers/eventRegistrations.spec.js

```diff
@@ -25,37 +23,236 @@ describe('Controller: eventRegistrations', function () {
+      expect(scope.builtInColumnsVisible).toEqual({
+        Email: true,
+        Group: true,
+        GroupId: false,
+        Started: true,
+        Completed: true,
+      });
+    });
+  });
+
+  it('toggleColumn toggles the column visibility', () => {
+    const setItem = spyOn($window.localStorage, 'setItem');
+    scope.toggleColumn(0);
+
+    expect(scope.blocks[0].visible).toBe(true);
+    expect(setItem).toHaveBeenCalledWith(
+      `visibleBlocks:${testData.conference.id}`,
+      '["e088fefc-eb9c-4904-b849-017facc9e063"]',
+    );
+
+    expect(scope.queryParameters.block).toEqual([
+      'e088fefc-eb9c-4904-b849-017facc9e063',
+    ]);
+
+    scope.toggleColumn(0);
+
+    expect(scope.blocks[0].visible).toBe(false);
+    expect(setItem).toHaveBeenCalledWith(
+      `visibleBlocks:${testData.conference.id}`,
+      '[]',
+    );
+
+    expect(scope.queryParameters.block).toEqual([]);
+  });
+
+  it('toggleBuiltInColumn toggles the column visibility', () => {
+    const setItem = spyOn($window.localStorage, 'setItem');
+    scope.toggleBuiltInColumn('Email');
+
+    expect(scope.builtInColumnsVisible.Email).toBe(false);
+    expect(setItem).toHaveBeenCalledWith(
+      `builtInColumnsVisibleStorage`,
+      '{"Email":false,"Group":true,"GroupId":false,"Started":true,"Completed":true}',
+    );
+
+    scope.toggleBuiltInColumn('Email');
+
+    expect(scope.builtInColumnsVisible.Email).toBe(true);
+    expect(setItem).toHaveBeenCalledWith(
+      `builtInColumnsVisibleStorage`,
+      '{"Email":true,"Group":true,"GroupId":false,"Started":true,"Completed":true}',
+    );
+  });
+
+  it('resetStrFilter clears the filter', () => {
+    scope.strFilter = 'Filter';
+    scope.resetStrFilter();
+
+    expect(scope.strFilter).toBe('');
+  });
+
+  describe('isRegistrantReported', () => {
+    it('returns true if the registrant is reported', () => {
+      scope.registrations = [{ ...testData.registration, reported: true }];
+
+      expect(
+        scope.isRegistrantReported(testData.registration.registrants[0]),
+      ).toBe(true);
+    });
+
+    it('returns false if the registrant is not reported', () => {
+      scope.registrations = [{ ...testData.registration, reported: false }];
+
+      expect(
+        scope.isRegistrantReported(testData.registration.registrants[0]),
+      ).toBe(false);
+    });
+  });
+
+  describe('answerSort', () => {
+    it('returns 0 if no orderBy is defined', () => {
+      scope.queryParameters.orderBy = undefined;
+
+      expect(scope.answerSort(testData.registration.registrants[0])).toBe(0);
+    });
+
+    it('returns "" if the answer does not exist', () => {
+      scope.queryParameters.orderBy = 'last_name';
+
+      expect(scope.answerSort(testData.registration.registrants[0])).toBe('');
+    });
+
+    it('returns the value of simple answers', () => {
+      scope.queryParameters.orderBy = '9b83eebd-b064-4edf-92d0-7982a330272a';
```

Comment:
As Bill was mentioning, we should prob make this into a variable, so it's easier to read and not mess up the test.